modules/internal/Bytes.chpl

```diff
@@ -0,0 +1,473 @@
+                to `count` times
+     */
+    proc replace(needle: bytes, replacement: bytes, count: int = -1) : bytes {
+
+    }
+
+    /*
+      Splits the :record:`bytes` on `sep` yielding the subbytes between each
+      occurrence, up to `maxsplit` times.
+
+      :arg sep: The delimiter used to break the object into chunks.
+      :arg maxsplit: The number of times to split the object, negative values
+                     indicate no limit.
+      :arg ignoreEmpty: * When `true`  -- Empty :record:`bytes` will not be
+                                          yielded, and will not count towards
+                                          `maxsplit`
+                        * When `false` -- Empty :record:`bytes` will be yielded
+                                          when `sep` occurs multiple times in a
+                                          row.
+     */
+    iter split(sep: bytes, maxsplit: int = -1,
+               ignoreEmpty: bool = false): bytes {
+
+    }
+
+    /*
+      Works as above, but uses runs of whitespace as the delimiter.
+
+      :arg maxsplit: The number of times to split the bytes, negative values
+                     indicate no limit.
+     */
+    iter split(maxsplit: int = -1) : bytes {
+
+    }
+
+    /*
+      Returns a new :record:`bytes` object, which is the concatenation of all of
+      the :record:`bytes` objects passed in with the receiving bytes inserted
+      between them.
+    */
+    proc join(const ref S: bytes ...) : bytes {
+
+    }
+
+    /*
+      Same as the varargs version, but with a homogeneous tuple of bytes.
+    */
+    proc join(const ref S) : bytes where isTuple(S) {
+
+    }
+
+    pragma "no doc"
+    proc join(ir: _iteratorRecord) {
+
+    }
+
+    /*
+      :arg chars: A :record:`bytes` containing each character to remove.
+                  Defaults to `" \\t\\r\\n"`.
+      :arg leading: Indicates if leading occurrences should be removed.
+                    Defaults to `true`.
+      :arg trailing: Indicates if trailing occurrences should be removed.
+                     Defaults to `true`.
+
+      :returns: A new :record:`bytes` with `leading` and/or `trailing`
+                occurrences of characters in `chars` removed as appropriate.
+    */
+    proc strip(chars: bytes = " \t\r\n", leading=true, trailing=true) : bytes {
+
+    }
+
+    /*
+      Splits the object on `sep` into a `3*bytes` consisting of the section
+      before `sep`, `sep`, and the section after `sep`. If `sep` is not found,
+      the tuple will contain the whole bytes, and then two empty bytes.
+    */
+    proc const partition(sep: bytes) : 3*bytes {
+
+    }
+
+    /*
+      Returns a UTF-8 string from the given :record:`bytes` object. If the data
+      is malformed for UTF-8, `errors` argument determines the action.
+      
+      :arg errors: `strict` raises an error, `replace` replaces the malformed
+                   data, `ignore` drops the data silently.
```

Comment:
"`replace` replaces the malformed data" -> "`replace` replaces the malformed character with the UTF-8 replacement character"

I'd expect such an argument to be an enum.